scipy/stats/stats.py

```diff
@@ -2772,11 +2780,15 @@ def spearmanr(a, b=None, axis=0):
 
     """
     a, axisout = _chk_asarray(a, axis)
+    if a.size <= 1:
+        return np.nan, np.nan
     ar = np.apply_along_axis(rankdata, axisout, a)
 
     br = None
     if b is not None:
         b, axisout = _chk_asarray(b, axis)
+        if b.size <= 1:
+            return np.nan, np.nan
```

Comment:
It's clearly documented that ``a`` and ``b`` need to be of the same length, so I'd prefer the current ``ValueError``.

Now ``stats.spearmanr([1, 2, 1], [8, 9])`` would raise an error but ``stats.spearmanr([1, 2, 1], 8) return ``nan, nan``.